seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -173,10 +187,18 @@ public class CoordinatorService {
 
     private PassiveCompletableFuture restoreAllJobFromMasterNodeSwitchFuture;
 
+    private ConcurrentLinkedQueue<Tuple2<Long, JobMaster>> pendingJob =
```

Comment:
Can we store jobid in jobmaster? Then invoke jobmaster::getJobId when we need jobid.